main.nf

```diff
@@ -615,145 +605,251 @@ process calculateNSCRSC {
+
+    script:
+    """
+    bamCoverage \\
+       -b $bam \\
+       --extendReads ${params.extendReadsLen} \\
+       --normalizeUsing RPKM \\
+       --numberOfProcessors ${task.cpus} \\
+       -o ${bam.baseName}.bw
+    """
+}
+
+/*
+ * STEP 7.4 deepTools computeMatrix
+ */
+process deepTools_computeMatrix {
+    tag "${bigwig[0].baseName}"
+    publishDir "${params.outdir}/deepTools", mode: 'copy'
+    label 'process_big'
+
+    input:
+    file bigwig from deepTools_bamCoverage_results.collect()
+    file bed from bed.collect()
+
+    output:
+    file 'computeMatrix.out.gz' into deepTools_computeMatrix_results
+
+    script:
+    """
+    computeMatrix scale-regions \\
+        --scoreFileName *.bw \\
+        --regionsFileName $bed \\
+        --beforeRegionStartLength 3000 \\
+        --afterRegionStartLength 3000 \\
+        --regionBodyLength 5000 \\
+        --outFileName computeMatrix.out.gz \\
+        --numberOfProcessors ${task.cpus} \\
+        --skipZeros \\
+        --smartLabels
+    """
+}
+
+
+/*
+ * STEP 7.5 deepTools computeMatrix
+ */
+process deepTools_plotProfile {
+    tag "${bam[0].baseName - '.dedup.sorted'}"
+    publishDir "${params.outdir}/deepTools", mode: 'copy'
+
+    input:
+    file bigwig from deepTools_computeMatrix_results
+
+    output:
+    file '*.{pdf,txt}' into deepTools_plotProfile_results
+    file '*.txt' into deepTools_plotProfile_multiqc
+
+    script:
+    """
+    plotProfile \\
+        --matrixFile computeMatrix.out.gz \\
+        --outFileName read_distribution_profile.pdf \\
+        --plotFileFormat pdf \\
+        --outFileNameData read_distribution_profile.txt \\
+        --plotTitle "Reads Distribution Profile"
+    """
+}
+
+
+/*
+ * STEP 7.6 deepTools multiBamSummary
+ */
+process deepTools_multiBamSummary {
+    tag "${bam[0].baseName - '.dedup.sorted'}"
+    publishDir "${params.outdir}/deepTools", mode: 'copy'
+    label 'process_big'
+
+    input:
+    file(bam) from bam_dedup_deepTools_multiBamSummary.filter( ~/.+\.bam$/ ).collect()
+
+    output:
+    file 'multiBamSummary.npz' into deepTools_multiBamSummary_results
+
+    when:
+    !(bam instanceof Path)
+
+    script:
+    """
+    multiBamSummary \\
+        bins \\
+        --binSize 10000 \\
+        --bamfiles $bam \\
+        -out multiBamSummary.npz \\
+        --extendReads ${params.extendReadsLen} \\
+        --ignoreDuplicates \\
+        --centerReads \\
+        --numberOfProcessors ${task.cpus} \\
+        --smartLabels
+    """
+}
+
+
+/*
+ * STEP 7.7 deepTools plotCorrelation
+ */
+process deepTools_plotCorrelation {
+    tag "${bam[0].baseName - '.dedup.sorted'}"
+    publishDir "${params.outdir}/deepTools", mode: 'copy'
+
+    input:
+    file npz from deepTools_multiBamSummary_results.collect()
+
+    output:
+    file '*.{pdf,txt}' into deepTools_plotCorrelation_results
+    file '*.txt' into deepTools_plotCorrelation_multiqc
+
+    when:
+    !(bam instanceof Path)
```

Comment:
What is this doing?

Reply:
Should stop it from running when there is only one file. Copied the logic from the earlier `if` statement.